src/Worker.php

```diff
@@ -64,11 +68,22 @@ private function assertEntityManagerIsOpen(): void
     /**
      * Pings the EntityManager's database connection to ensure that it is still open. If the connection is not open,
      * this method will attempt to re-open the connection.
+     *
+     * @throws Exception
      */
     private function ensureDatabaseConnectionIsOpen(): void
     {
         $connection = $this->entityManager->getConnection();
-        if (!$connection->ping()) {
+
+        // This replicates what the deprecated ping() function used to do.
+        try {
+            $connection->executeQuery($connection->getDatabasePlatform()->getDummySelectSQL());
+            $ping = true;
+        } catch (Exception) {
```

Comment:
I think this throws a `ConnectionLost` if the ping fails? Probably better to catch that specific exception and let any other random errors bubble up.

Reply:
It only throws a `Doctrine\DBAL\Exception`. They updated it the exception.

`ping()` looked like this
![Screenshot 2024-02-02 at 9 51 37 AM](https://github.com/pinnacleqi/doctrine-queue-worker/assets/57735637/1ba9e347-86a8-41fe-a77b-35def423f668)

Reply:
What does `executeQuery()` throw if the query fails due to the connection timing out or being closed?

Reply:
It can throw a `Doctrine\DBAL\Driver\Exception` (which can be an instance of ConnectionLost), but it catches it and handles it.

![Screenshot 2024-02-02 at 10 14 32 AM](https://github.com/pinnacleqi/doctrine-queue-worker/assets/57735637/13893601-5dc2-4ec8-81eb-4c81b112fe54)
![Screenshot 2024-02-02 at 10 13 26 AM](https://github.com/pinnacleqi/doctrine-queue-worker/assets/57735637/af2bca70-9fdf-42f7-ad24-7e2379e97ca4)

Technically `executeQuery()` can also throw a `Doctrine\DBAL\Cache\CacheException`, which isn't documented in the function's throw tags. The only throw they have documented is the `Doctrine\DBAL\Exception`

Reply:
So I guess it technically can throw the `Doctrine\DBAL\Driver\Exception` too. Which can be an instance of ConnectionLost.
Should I account for that in the catch, as well?

Reply:
Don't we want it to stay as close as possible to the previous functionality, though?

Reply:
Ah, I was assuming we were catching PHP's root `\Exception`, but it looks like we're importing `Doctrine\DBAL\Exception`, so this code appears to be doing the same thing `ping()` did, and we're all good with the way you wrote it.